server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -135,25 +132,36 @@ private void countDown() {
             remoteRequest.runtimeFields(request.runtimeFields());
             remoteRequest.indexFilter(request.indexFilter());
             remoteRequest.nowInMillis(nowInMillis);
-            remoteClusterClient.fieldCaps(remoteRequest,
-                ActionListener.wrap(response -> {
-                    List<FieldCapabilitiesIndexResponse> remotes = new ArrayList<>();
-                    for (FieldCapabilitiesIndexResponse resp : response.getIndexResponses()) {
-                        remotes.add(new FieldCapabilitiesIndexResponse(
+            remoteClusterClient.fieldCaps(remoteRequest, ActionListener.wrap(response -> {
+                List<FieldCapabilitiesIndexResponse> remotes = new ArrayList<>();
+                for (FieldCapabilitiesIndexResponse resp : response.getIndexResponses()) {
+                    indexResponses.add(
+                        new FieldCapabilitiesIndexResponse(
                             RemoteClusterAware.buildRemoteIndexName(clusterAlias, resp.getIndexName()),
-                            resp.get(), resp.canMatch()));
-                    }
-                    countDownListener.onResponse(remotes);
-                }, countDownListener::onFailure));
+                            resp.get(),
+                            resp.canMatch()
+                        )
+                    );
+                }
+                for (Map.Entry<String, Exception> entry : response.getFailures().entrySet()) {
+                    indexFailures.put(RemoteClusterAware.buildRemoteIndexName(clusterAlias, entry.getKey()), entry.getValue());
+                }
+                countDown.run();
+            }, e -> {
+                // individual index failures should show up as remote responses containing an exception
+                countDown.run();
```

Comment:
This case is not handled automatically. We need to add all the indices from that cluster in the index failures map. Since all indices on that cluster failed, maybe that the key in the map could be the cluster name instead ?

Reply:
Individual index failures on a remote should be handled (there is also a test for it now that should fail otherwise). So you mean this should cover general remote failures? How could this happen? Also asking since I cannot thing around a way of testing this atm...